src/main/java/org/apache/cassandra/sidecar/cdc/CDCReaderService.java

```diff
@@ -0,0 +1,157 @@
+package org.apache.cassandra.sidecar.cdc;
+
+import java.io.InvalidObjectException;
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Host;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.apache.cassandra.schema.KeyspaceMetadata;
+import org.apache.cassandra.schema.Schema;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.sidecar.CQLSession;
+
+/**
+ * Cassandra's real-time change data capture service.
+ */
+@Singleton
+public class CDCReaderService implements Host.StateListener
+{
+    private static final Logger logger = LoggerFactory.getLogger(CDCReaderService.class);
+    private final CDCIndexWatcher cdcIndexWatcher;
+    private final CDCRawDirectoryMonitor cdcRawDirectoryMonitor;
+    private final CQLSession session;
+    private final CassandraConfig cassandraConfig;
+
+    @Inject
+    public CDCReaderService(CDCIndexWatcher cdcIndexWatcher, CDCRawDirectoryMonitor monitor, CQLSession session,
+                            CassandraConfig cassandraConfig)
+    {
+        this.cdcRawDirectoryMonitor = monitor;
+        this.cdcIndexWatcher = cdcIndexWatcher;
+        this.session = session;
+        this.cassandraConfig = cassandraConfig;
+    }
+
+    public synchronized void start()
+    {
+        try
+        {
+            // Wait until the Cassandra server is UP to load configs and start subsystems. There's no guarantee the
+            // config is valid otherwise.
+            waitForCassandraServer();
+            Cluster cluster = session.getLocalCql().getCluster();
+            if (cluster == null)
+            {
+                throw new InvalidObjectException("Cannot connect to the local Cassandra node");
+            }
+
+            // Ensure Cassandra config is valid and remove mutable data paths from the config
+            // to ensure CDC reader doesn't accidentally step on Cassandra data.
+            this.cassandraConfig.init();
+            // TODO : Load metadata from the CQLSession.
+            Schema.instance.loadFromDisk(false);
+            this.cassandraConfig.muteConfigs();
+
+            for (String keySpace : Schema.instance.getKeyspaces())
+            {
+                logger.info("Keyspace : {}", keySpace);
+                KeyspaceMetadata keyspaceMetadata = Schema.instance.getKeyspaceMetadata(keySpace);
+                if (keyspaceMetadata == null)
+                {
+                    continue;
+                }
+                for (TableMetadata tableMetadata : keyspaceMetadata.tablesAndViews())
+                {
+                    logger.info("Table : {}, CDC enabled ? {}", tableMetadata.name,
```

Comment:
After read the code, I saw that this code is useless ,If the cdc is enable can get from mutation at the mutationhandler section . So the code here is useless I think ,And we may not doing the schema load disk method to get the schema meta data

Reply:
Yes, this is just a dummy code for interim development work. I will remove or move these to debug logs.